### DISCUSSION TITLE
Unable to Download Devbox Template or Recover Code when Frozen

### DISCUSSION THREAD

**Discussion by DiegoFleitas** [codesandbox/codesandbox-client#8679] [Q&A]
I accidentally left a Node.js Devbox running overnight, which used up all my credits. Now my workspace is frozen, and I can't access or download the code I was experimenting with. I created a template to work around the issue, but the template is locked into the Devbox system, and Sandboxes are not available for it.
Questions

- Is there a way to download the code from my template without requiring credits?
- Can templates created from Devboxes be converted into a format compatible with Sandboxes?
- Is there an alternative way to access or recover my experiment, such as a temporary unlock or export option?

Why This is a Problem

1.  Templates should be reusable and accessible, but they are tied to the Devbox runtime.
2.  No warning was provided that leaving a Devbox running would completely freeze access to the Node fiddle I was working on.
3.  There’s no option to download the template code as a ZIP or view it in read-only mode without credits.
4.  While the code I was fiddling with isn’t that important, the design of this system feels backwards.

Feedback

A simple export option for templates would resolve this issue entirely.

This experience makes me feel pressured to upgrade to a Pro subscription, which is disappointing. I’d prefer a simple way to download my experiment without restrictions, as that would make the platform feel more user-friendly and reliable for future use.
If I had known this would happen, I wouldn’t have used Devboxes in the first place—I would have stuck to JS Sandboxes instead...

Is this deliberate design, or am I missing something?

**Comment by filipelima18** [comment]
Hi @DiegoFleitas, Thanks for opening this. We are working to improve the visibility of VM credits and provide additional warnings when a Free workspace is approaching the allocated monthly VM credits.

While we are working on this, please email support@codesandbox.io and we can help you download your work.

**Reply by DiegoFleitas** [reply]
No thanks, I paid the price, downloaded my code and cancelled right after.
I won't be using this... It looked cool but I won't stand for a platform that uses your own code as a leverage to get you to go premium, it doesn't sit right with me

**Reply by filipelima18** [reply]
Thanks for sharing this feedback with us. We are working to ensure that the Free experience is more transparent and gives more visibility over credit usage.
